generator/internal/sidekick/sidekick.go

```diff
@@ -17,57 +17,95 @@ package sidekick
 
 import (
 	"fmt"
-	"maps"
 	"os"
+	"strings"
 )
 
-func Run() error {
-	cmdLine, err := parseArgs()
-	if err != nil {
-		return err
+var cmdSidekick = newCommand(
+	"sidekick",
+	"sidekick is a tool for automating SDK generation.",
+	``,
+	nil, // nil parent is only allowed for the root command
+	nil).
+	addFlagString(&flagProjectRoot, "project-root", "", "the root of the output project").
+	addFlagString(&format, "specification-format", "", "the specification format. Protobuf or OpenAPI v3.").
+	addFlagString(&source, "specification-source", "", "the path to the input data").
+	addFlagString(&serviceConfig, "service-config", "", "path to service config").
+	addFlagString(&output, "output", "", "the path within project-root to put generated files").
+	addFlagString(&flagLanguage, "language", "rust", "the generated language").
```

Comment:
I made `rust` the default value for `-language` flag because unparam wasn't happy with this value being always empty. I couldn't manage to fix it with `//nolint:unparam`, and didn't think it was worth removing the parameter just to make it happy. For the time being, `rust` is probably the right value for this flag anyways :)